### PR TITLE
VIM-5583: Add `VIMLiveHeartbeat` Class

### PR DESCRIPTION
**Pull request by supperspidey**
#### Ticket

[VIM-5583](https://vimean.atlassian.net/browse/VIM-5583)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

The goal of this PR is to add a new class representing the `live` field in a `file` response.

#### Implementation Summary

N/A

#### Reviewer Tips

N/A

#### How to Test

- Run all tests and make sure they pass.

### CODE REVIEW

File: VimeoNetworking/Sources/Models/VIMLiveHeartbeat.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// An object representing the `live` object in either an `hls` or a `dash` response.
+public class VIMLiveHeartbeat: VIMModelObject
```
Comment (jasonhawkins) [inline VimeoNetworking/Sources/Models/VIMLiveHeartbeat.swift]: We should be able to get away with just calling this `LiveHeartbeat`, right? I ended up calling our play progress object just `PlayProgress` which didn't seem to cause any name space issues. Maybe it's not worth worrying about now though.

File: VimeoNetworking/Sources/Models/VIMVideoPlayFile.h [hunk 2]
```diff
@@ -28,11 +28,14 @@
...
 #import "VIMModelObject.h"
 
+@class VIMLiveHeartbeat;
+
 @interface VIMVideoPlayFile : VIMModelObject
 
 @property (nonatomic, copy, nullable) NSString *link;
 @property (nonatomic, strong, nullable) NSDate *expirationDate;
 @property (nonatomic, copy, nullable) NSString *log;
+@property (nonatomic, strong, nullable) VIMLiveHeartbeat *live;
```
Comment (jasonhawkins) [inline VimeoNetworking/Sources/Models/VIMVideoPlayFile.h]: Can we call this `heartbeat` or `liveHeartbeat` instead?

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOSTests/VIMLiveHeartbeatTests.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                let liveHeartbeat = video.playRepresentation?.hlsFile?.live
+                
+                XCTAssertNotNil(liveHeartbeat)
+                XCTAssertEqual(liveHeartbeat?.heartbeatUrl, "https://api.vimeo.com/videos/236661678/heartbeat/hls/1507067883/3082903e82078a78bd146ca542899a124e1c47fe")
+                
+            case .failure(let error):
+                XCTFail("\(error)")
+            }
+            
+            expectation.fulfill()
```
Comment (jasonhawkins) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOSTests/VIMLiveHeartbeatTests.swift]: ⭐️

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          12 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/VimeoNetworking/Sources/VimeoClient.swift#L465">VimeoNetworking/Sources/VimeoClient.swift#L465</a></strong>: redundant conformance constraint ‘ModelType’: ‘MappableResponse’  <br /><code>
    private func handleError&lt;ModelType: MappableResponse&gt;(_ error: NSError, request: Request&lt;ModelType&gt;, task: URLSessionDataTask? = nil)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L97">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L97</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L106">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L106</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L75">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L75</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L95">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L95</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L104">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L104</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L76">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L76</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/VimeoNetworking/Sources/VimeoRequestSerializer.swift#L43">VimeoNetworking/Sources/VimeoRequestSerializer.swift#L43</a></strong>: when calling this function in Swift 4 or later, you must pass a ‘()’ tuple; did you mean for the input type to be ‘()’?  <br /><code>
    public typealias AccessTokenProvider = (Void) -&gt; String?
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/VimeoNetworking/Sources/VimeoClient.swift#L177">VimeoNetworking/Sources/VimeoClient.swift#L177</a></strong>: redundant conformance constraint ‘ModelType’: ‘MappableResponse’  <br /><code>
    public func request&lt;ModelType: MappableResponse&gt;(_ request: Request&lt;ModelType&gt;, completionQueue: DispatchQueue = DispatchQueue.main, completion: @escaping ResultCompletion&lt;Response&lt;ModelType&gt;&gt;.T) -&gt; RequestToken
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/VimeoNetworking/Sources/VimeoClient.swift#L320">VimeoNetworking/Sources/VimeoClient.swift#L320</a></strong>: redundant conformance constraint ‘ModelType’: ‘MappableResponse’  <br /><code>
    private func handleTaskSuccess&lt;ModelType: MappableResponse&gt;(forRequest request: Request&lt;ModelType&gt;, task: URLSessionDataTask?, responseObject: Any?, isCachedResponse: Bool = false, isFinalResponse: Bool = true, completionQueue: DispatchQueue, completion: @escaping ResultCompletion&lt;Response&lt;ModelType&gt;&gt;.T)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/VimeoNetworking/Sources/VimeoClient.swift#L425">VimeoNetworking/Sources/VimeoClient.swift#L425</a></strong>: redundant conformance constraint ‘ModelType’: ‘MappableResponse’  <br /><code>
    private func handleTaskFailure&lt;ModelType: MappableResponse&gt;(forRequest request: Request&lt;ModelType&gt;, task: URLSessionDataTask?, error: NSError?, completionQueue: DispatchQueue, completion: @escaping ResultCompletion&lt;Response&lt;ModelType&gt;&gt;.T)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/734f518fc29ac8afa17878cb3453c5aa9c599723/VimeoNetworking/Sources/ExceptionCatcher+Swift.swift#L38">VimeoNetworking/Sources/ExceptionCatcher+Swift.swift#L38</a></strong>: when calling this function in Swift 4 or later, you must pass a ‘()’ tuple; did you mean for the input type to be ‘()’?  <br /><code>
    @nonobjc internal static func doUnsafe(unsafeBlock: @escaping ((Void) -&gt; Void)) throws
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 71 tests, with 0 failures (0 unexpected) in 0.758 (0.814) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/529)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**